### PR TITLE
General components

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: metro.config.js [hunk 1]
```diff
@@ -5,13 +5,37 @@
...
-        inlineRequires: true,
-      },
-    }),
-  },
-};
+// module.exports = {
+//   transformer: {
+//     getTransformOptions: async () => ({
+//       transform: {
+//         experimentalImportSupport: false,
```
Comment (chengoM123) [inline metro.config.js]: remove comments

File: metro.config.js [hunk 2]
```diff
@@ -5,13 +5,37 @@
...
+  const {
+    resolver: { sourceExts, assetExts }
+  } = await getDefaultConfig()
+  return {
+    transformer: {
+      babelTransformerPath: require.resolve("react-native-svg-transformer"),
+      // getTransformOptions: async () => ({
+      //   transform: {
+      //     experimentalImportSupport: false,
+      //     inlineRequires: true,
```
Comment (chengoM123) [inline metro.config.js]: remove comments

File: src/components/common/AppInput.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+                            : <Hidden />}
+                    </Pressable>
+                </View>
+            </View>
+            {error && <Text style={styles.error}>{error}</Text>}
+        </Fragment>
+    )
+
+    else if (contentType === ContentType.email) return (
+        <Fragment>
```
Comment (chengoM123) [inline src/components/common/AppInput.tsx]: use view instead fragment

File: src/components/common/AppInput.tsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+            setBorderColor(Colors.red500)
+        }
+        else {
+            setError('')
+            setBorderColor(Colors.gray600)
+        }
+    }
+
+    if (contentType === ContentType.password) return (
+        <Fragment>
```
Comment (chengoM123) [inline src/components/common/AppInput.tsx]: use view instead fragment

File: src/components/common/LogisterButton.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+function LogisterButton({ children, onPress, textStyle, bgColor, icon }: LogisterButtonProps) {
+    return (
+        <View style={styles.buttonOuterContainer}>
+            <Pressable
+                onPress={onPress}
+                style={
+                    ({ pressed }) => pressed
+                        ? [styles.pressed, styles.buttonInnerContainer, { backgroundColor: bgColor }]
```
Comment (chengoM123) [inline src/components/common/LogisterButton.tsx:18]: add the backgroundColor to the pressed style

File: src/constants/colors.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+export const Colors = {
+    gray500: '#F1F1F9',
```
Comment (chengoM123) [inline src/constants/colors.ts]: capital letters

File: src/utils/functions.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
+export const validatePassword = (password: string): string | false => {
+    const regexLower = /^(?=.*[a-z]).+$/
```
Comment (chengoM123) [inline src/utils/functions.ts]: Rename the file to "validationUtils"

File: src/components/common/LogisterButton.tsx (hunk 5 above)
Comment (ghost) [reply src/components/common/LogisterButton.tsx:18]: it is already added in the end of the array, isn't it?

File: src/constants/colors.ts (hunk 6 above)
Comment (ghost) [reply src/constants/colors.ts]: do you mean capital letters for the Colors variable or for the gray500 key? or both?

File: App.tsx [hunk 8]
```diff
@@ -1,35 +1,90 @@
...
+import LogisterButton from './src/components/common/LogisterButton'
+import { validateEmail, validatePassword } from './src/utils/validationUtils'
+import ArrowRight from './assets/arrow-right.svg'
 
 const App = () => {
+  const [value, setValue] = React.useState('')
+  const [value2, setValue2] = React.useState('')
   return (
-    <>
+    <Fragment>
```
Comment (chengoM123) [inline App.tsx]: you dont need the fragment

File: App.tsx [hunk 9]
```diff
@@ -1,35 +1,90 @@
...
         <ScrollView
           contentInsetAdjustmentBehavior="automatic"
           style={styles.scrollView}>
-          <Header />
-          <View style={styles.body}>
-            <Counter />
-            <LearnMoreLinks />
+          <View style={styles.test}>
+            <AppInput
+              placeholderText='this is a passowrd'
```
Comment (chengoM123) [inline App.tsx]: use strings file

**Review by chengoM123 — APPROVED** [review]

File: src/constants/colors.ts (hunk 6 above)
Comment (chengoM123) [reply src/constants/colors.ts]: for the key

File: src/components/common/LogisterButton.tsx (hunk 5 above)
Comment (chengoM123) [reply src/components/common/LogisterButton.tsx:18]: yes, but we prefer not to use inline style.
you can add this to the pressed style object

**Pull request merged**